pkg/apis/core/types.go

```diff
@@ -3776,6 +3779,27 @@ const (
 // ResourceList is a set of (resource name, quantity) pairs.
 type ResourceList map[ResourceName]resource.Quantity
 
+func (r *ResourceList) UnmarshalJSONPB(u *jsonpb.Unmarshaler, value []byte) error {
+	var m map[string]interface{}
+	err := json.Unmarshal(value, &m)
+	if err != nil {
+		return err
+	}
+
+	resourceList := make(ResourceList)
+	for resourceName, quantity := range m {
+		quantityMap := quantity.(map[string]interface{})
+		quantityIface, ok := quantityMap["string"]
```

Comment:
Should you do this? or just pass it along to json Unmarshal Quantity and let it handle that?

Reply:
good tip, just passed it along